examples/address_generator/src/main.rs

```diff
@@ -0,0 +1,134 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use clap::{App, Arg};
+
+use transact::contract::address::{
+    double_key_hash::DoubleKeyHashAddresser, key_hash::KeyHashAddresser,
+    triple_key_hash::TripleKeyHashAddresser, Addresser,
+};
+
+fn main() -> Result<(), String> {
+    let matches = App::new(clap::crate_name!())
+        .version(clap::crate_version!())
+        .author(clap::crate_authors!())
+        .about(clap::crate_description!())
+        .arg(
+            Arg::with_name("prefix")
+                .required(true)
+                .takes_value(true)
+                .help("Prefix of the radix address.")
+                .index(1),
+        )
+        .arg(
+            Arg::with_name("key")
+                .required(true)
+                .multiple(true)
+                .takes_value(true)
+                .max_values(3)
+                .index(2)
+                .help(
+                    "Natural key used to compute the radix address.\nMust supply at least 1 key.",
+                ),
+        )
+        .arg(
+            Arg::with_name("length")
+                .required(false)
+                .max_values(2)
+                .last(true)
+                .index(3)
+                .help(
+                    "Optional length of hashes used to construct the radix address.\nMust be u32.",
+                ),
+        )
+        .get_matches();
+
+    // The `prefix` will be used to construct the Addresser
+    let prefix = matches
+        .value_of("prefix")
+        .expect("Unable to get `prefix` arg")
+        .to_string();
+
+    let keys: Vec<_> = matches
+        .values_of("key")
+        .expect("Unable to get `key` arg")
+        .collect();
+
+    // Optional lengths used in the construction of the radix address
+    let lengths = match matches.values_of("length") {
+        Some(vals) => vals
+            .map(|l| {
+                l.parse()
+                    .map_err(|_| "Unable to parse `length` arg".to_string())
+            })
+            .collect::<Result<Vec<usize>, String>>()?,
+        _ => vec![],
+    };
+
+    // The Addresser implementation to be constructed depends on the amount of `keys` supplied.
+    // Once the appropriate Addresser implementation is constructed, the radix address will be
+    // calculated and printed out.
+    match keys.len() {
+        1 => {
+            let addresser = KeyHashAddresser::new(prefix);
+            let addr = addresser
+                .compute(&keys[0].to_string())
+                .map_err(|e| format!("Unable to compute address with KeyHashAddresser: {}", e))?;
```

Comment:
Remove reference to KeyHashAddresser here. The CLI user isn't familiar with the code.

Reply:
Same below for DoubleKeyHashAddresser and TripleKeyHashAddresser